scripts/dev/upstream_merge/build_and_install_kernel.py

```diff
@@ -0,0 +1,291 @@
+        env = os.environ.copy()
+        if config.arch:
+            env["ARCH"] = config.arch
+        env["CROSS_COMPILE"] = CROSS_COMPILE
+        if not args.skip_merge:
+            run_upstream_merge_script()
+        else:
+            ensure_kernel_source_repository()
+        kernel_src_dir = config.kernel_src_dir
+        print("[INFO] Cleaning previous builds...")
+        if not args.dry_run:
+            status, output = run_command("make mrproper", cwd=kernel_src_dir, env=env)
+            if status != 0:
+                raise RuntimeError(output)
+        kernel_config_target = config.kernel_config or "defconfig"
+        print(f"[INFO] Creating kernel configuration: {kernel_config_target}...")
+        if not args.dry_run:
+            status, output = run_command(f"make {kernel_config_target}", cwd=kernel_src_dir, env=env)
+            if status != 0:
+                raise RuntimeError(output)
+        make_jobs = config.make_jobs or "$(nproc)"
+        if make_jobs == "$(nproc)":
+            make_jobs = str(os.cpu_count())
+        print(f"[INFO] Building kernel and modules with {make_jobs} jobs...")
+        if not args.dry_run:
+            status, output = run_command(f"make -j{make_jobs} bzImage modules", cwd=kernel_src_dir, env=env)
+            if status != 0:
+                raise RuntimeError(output)
+        temp_modules_dir = config.temp_modules_dir or os.path.join(kernel_src_dir, "tmp-modules")
+        print(f"[INFO] Staging modules in: {temp_modules_dir}...")
+        if not args.dry_run:
+            if os.path.exists(temp_modules_dir):
+                shutil.rmtree(temp_modules_dir)
+            os.makedirs(temp_modules_dir, exist_ok=True)
+            status, output = run_command(f"make modules_install INSTALL_MOD_PATH={temp_modules_dir}", cwd=kernel_src_dir, env=env)
+            if status != 0:
+                raise RuntimeError(output)
+        kernel_version = run_command("make -s kernelrelease", cwd=kernel_src_dir, env=env)[1] if not args.dry_run else "DRY-RUN-VERSION"
+        target_host = config.kernel_target_host or config.rt_target_IP
+        target_user = config.kernel_target_user or "admin"
+        if not target_host:
+            raise RuntimeError("[ERROR] No target host configured")
+        print(f"[INFO] Copying kernel to target {target_host}...")
+        if not args.dry_run:
+            status, output = run_command(f"scp {kernel_src_dir}/arch/x86/boot/bzImage {target_user}@{target_host}:/boot/bzImage-{kernel_version}")
+            if status != 0:
+                raise RuntimeError(output)
+        print("[INFO] Backing up existing kernel on target (if needed)...")
+        if not args.dry_run:
+            status, output = run_command(f"ssh {target_user}@{target_host} 'if [ -f /boot/runmode/bzImage ] && [ ! -h /boot/runmode/bzImage ]; then mv /boot/runmode/bzImage /boot/runmode/bzImage-$(uname -r); else echo Skipping backup; fi'")
```

Comment:
You might want to run the backup step before actually copying the kernel over to the target.